app/grandchallenge/components/tasks.py

```diff
@@ -279,13 +347,37 @@ def remove_container_image_from_registry(
 
 @acks_late_2xlarge_task(ignore_errors=(InstanceInUse,))
 def delete_container_image(*, pk: uuid.UUID, app_label: str, model_name: str):
+    from grandchallenge.algorithms.models import AlgorithmImage, Job
+    from grandchallenge.evaluation.models import Evaluation, Method
+    from grandchallenge.workstations.models import WorkstationImage
+
     remove_container_image_from_registry(
         pk=pk, app_label=app_label, model_name=model_name
     )
 
     model = apps.get_model(app_label=app_label, model_name=model_name)
     instance = model.objects.get(pk=pk)
 
+    if isinstance(instance, Method):
+        should_be_protected = Evaluation.objects.filter(
+            method=instance,
+            status=Evaluation.SUCCESS,
+        ).exists()
+    elif isinstance(instance, AlgorithmImage):
+        should_be_protected = Job.objects.filter(
+            algorithm_image=instance,
+            status=Job.SUCCESS,
+        ).exists()
+    elif isinstance(instance, WorkstationImage):
+        should_be_protected = instance.created > (
+            now() - relativedelta(years=1)
+        )
```

Comment:
Suggestion: Would it make sense to also spare the "active image"? Maybe has been though about, I do not know...

Reply:
If it is over a year old then the license has expired and it will not run.